logrotate.c

```diff
@@ -2798,8 +2798,10 @@ static int writeState(const char *stateFilename)
     }
     strcpy(tmpFilename, stateFilename);
     strcat(tmpFilename, ".tmp");
-    /* Remove possible tmp state file from previous run */
-    error = unlink(tmpFilename);
+    /* If present, remove possible tmp state file from previous run */
```

Comment:
On the other hand, this introduces [TOCTOU](https://cwe.mitre.org/data/definitions/367.html), which will trigger static analyzers.  It is not an improvement really.